text/####-error-mitigation.md

```diff
@@ -0,0 +1,235 @@
+            "description": "The stretch factor of the gate for error mitigation (if supported)."
+        }
+    }
+}
+```
+To facilitate the user experience, helper functions that show the available stretch factors, will be created.
+First, we define that a given qubit has stretch factor `c` available if all gates that involve this qubit are available with this stretch factor.
+The user will thus be able to query the backend to find out which stretch factors are available on a given list of qubits:
+```
+def available_stretch_factors(self, qubits: List) -> List:
+    """Returns a list of stretch factors available for the given qubits."""
+```
+For example to find the stretch factors available to qubits 0, 1, and 3, the user would do `backend.available_stretch_factors([0, 1, 3])`.
+Similarly, the user may query the backend to find out which qubits support a given list of stretch factors.
+```
+def available_qubits(self, stretch_factors: List) -> List:
+    """Returns a list of qubits for which the given stretch factors are available."""
+```
+The user may also query the backend to find out how the error mitigation is configured.
+```
+def richardson_error_mitigation_config(self) -> List:
+    """Returns a List of dict showing the available configuration of the richardson error mitigation."""
+```
+The returned list would contain dictionaries of the form
+```
+{'stretch_factors': list, 'qubits': list}.
+```
+For example, the list
+```
+[
+  {'stretch_factors': ['1.0', '1.1', '1.25'], 'qubits': [0, 1, 2, 3]},
+  {'stretch_factors': ['1.0', '1.1'], 'qubits': [0, 1, 2, 3, 4]}
+]
+```
+would imply that qubit 4 does not have stretch factor 1.25 available.
+
+To execute a quantum circuit with error mitigation, the user would do
+```
+execute(circ, backend, ..., error_mitigation='richardson')
+```
+to use all the stretch factors available to the qubits in the quantum register of `circ`.
+Additionally, the user may specify a subset of stretch factors to use by doing
+```
+execute(circ, backend, ..., error_mitigation='richardson', stretch_factors=[1.0, 1.25, 1.5]).
+```
+This will require a validation step, done in Qiskit Terra and based on the `config`, to check that the specified stretch factors are available for all of the qubits in the quantum register of `circ`.
+If this is not the case, an error will be raised.
+Here, `error_mitigation` specifies the error mitigation method to use.
+This value would by default be `None` when error mitigation is not used.
+The changes needed in Qiskit to implement error mitigation would require changes in the `assemble` function.
+`assemble` would detect that the user requires error mitigation and replace each circuit by several circuits, one for each stretch factor.
+The `experiments` in the resulting `qobj` would have an entry in the header that specifies the stretch factor.
+For example, the `qobj` for two circuits (`circA` and `circB`) to be run with error mitigation with three stretch factors would look like
```

Comment:
```suggestion
For example, the `qobj` for two circuits (`circA` and `circB`) to be run with error mitigation with three stretch factors would include six experiments, four of which would look like
```

Reply:
@ajavadia Is this what you were getting at in the question below?

Reply:
yes